test/jsx-import.js

```diff
@@ -0,0 +1,201 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import arrayToSentence from 'array-to-sentence';
+import rule from '../rules/jsx-import';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	},
+	parserOptions: {
+		ecmaFeatures: {
+			jsx: true
+		},
+		sourceType: 'module'
+	}
+});
+
+const pragmas = {
+	h: ['dom-chef', 'preact'],
+	React: ['react'],
+	preact: ['preact']
+};
+
+const suggestedModules = name => arrayToSentence(pragmas[name].map(x => `\`${x}\``), {lastSeparator: ' or '});
+
+const missingError = {
+	ruleId: 'jsx-import',
+	message: 'A valid pragma must be in scope when using JSX'
+};
+
+const superflousError = name => ({
+	ruleId: 'jsx-import',
+	message: `\`${name}\` shouldn't be imported when not using JSX`
+});
+
+const wrongError = name => ({
+	ruleId: 'jsx-import',
+	message: `\`${name}\` should be imported from ${suggestedModules(name)}`
+});
+
+ruleTester.run('jsx-import', rule, {
+	valid: [
+		'import React from \'react\'; let Foo; <Foo/>;',
+		'import preact from \'preact\'; let Foo; <Foo/>;',
+		'import preact, {render} from \'preact\'; let Foo; <Foo/>;',
+		'import {h, render} from \'preact\'; let Foo; <Foo/>;',
+		'import {h} from \'dom-chef\'; let Foo; <Foo/>;',
+		'import React from \'react\'; import bar from \'bar\'; let Foo; <Foo/>;',
+		'const React = require(\'react\'); let Foo; <Foo/>;',
+		'const preact = require(\'preact\'); let Foo; <Foo/>;',
+		'const {h} = require(\'preact\'); let Foo; <Foo/>;',
+		'const {h} = require(\'dom-chef\'); let Foo; <Foo/>;',
+		'const {h} = require(\'dom-chef\'); const bar = require(\'bar\'); let Foo; <Foo/>;',
```

Comment:
`dom-chef` supports using `import React from \'dom-chef\';` now, so that should be tested.

Reply:
@sindresorhus If I add `dom-chef` to support React then when using below code won't be fixable.

`import React from 'foobar'`

Reply:
I think that's fine. Nobody would do `import React from 'foobar'` any way.